### PR TITLE
Support for W5500-EVB-Pico board

### PR DESCRIPTION
**Pull request by michi-jung**
## Summary

This pull request contains support for the WIZnet W5500-EVB-Pico bord.

## Impact

Should not have any impact on other board configurations.

## Testing

Tested this on a W5500-EVB-Pico including Ethernet with TCP/IPv4 & IPv6.

### CODE REVIEW

**Comment by acassis** [comment]
Nice work @michi-jung !!!

Could you please this error reported in the network stack:

```
In file included from dhcp6c.c:37:
dhcp6c.c: In function 'dhcp6c_get_result':
Error: dhcp6c.c:435:9: error: format '%d' expects argument of type 'int', but argument 3 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
  435 |   ninfo("T1:%d T2:%d for iface %i\n", presult->t1, presult->t2,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~
      |                                              |
      |                                              uint32_t {aka long unsigned int}
dhcp6c.c:435:14: note: format string is defined here
  435 |   ninfo("T1:%d T2:%d for iface %i\n", presult->t1, presult->t2,
      |             ~^
      |              |
      |              int
      |             %ld
Error: dhcp6c.c:435:9: error: format '%d' expects argument of type 'int', but argument 4 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
  435 |   ninfo("T1:%d T2:%d for iface %i\n", presult->t1, presult->t2,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~               ~~~~~~~~~~~
      |                                                           |
      |                                                           uint32_t {aka long unsigned int}
dhcp6c.c:435:20: note: format string is defined here
  435 |   ninfo("T1:%d T2:%d for iface %i\n", presult->t1, presult->t2,
      |                   ~^
      |                    |
      |                    int
      |                   %ld
dhcp6c.c: In function 'dhcp6c_command':
Error: dhcp6c.c:796:9: error: format '%u' expects argument of type 'unsigned int', but argument 4 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
  796 |   ninfo("Sending %s (timeout %u s)\n", retx->name, timeout);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~              ~~~~~~~
      |                                                    |
      |                                                    uint32_t {aka long unsigned int}
dhcp6c.c:796:31: note: format string is defined here
  796 |   ninfo("Sending %s (timeout %u s)\n", retx->name, timeout);
      |                              ~^
      |                               |
      |                               unsigned int
      |                              %lu
```

It is just replacing %u with PRIu32

File: boards/arm/rp2040/w5500-evb-pico/include/board.h [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: rp2040_boardearlyinitialize
+ *
+ * Description:
+ *
```
Comment (acassis) [inline boards/arm/rp2040/w5500-evb-pico/include/board.h:148]: Please include description

File: boards/arm/rp2040/w5500-evb-pico/include/board.h [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+ ****************************************************************************/
+
+void rp2040_boardearlyinitialize(void);
+
+/****************************************************************************
+ * Name: rp2040_boardinitialize
+ *
+ * Description:
+ *
```
Comment (acassis) [inline boards/arm/rp2040/w5500-evb-pico/include/board.h:157]: Ditto

File: boards/arm/rp2040/w5500-evb-pico/include/rp2040_i2cdev.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: board_i2cdev_initialize
+ *
+ * Description:
+ *   Initialize i2c driver and register the /dev/i2c device.
+ *
```
Comment (acassis) [inline boards/arm/rp2040/w5500-evb-pico/include/rp2040_i2cdev.h:59]: Please include the Input parameter: bus

**Comment by acassis** [comment]
@michi-jung also please include the board documentation with its picture, see reference:
https://nuttx.apache.org/docs/latest/platforms/arm/rp2040/boards/raspberrypi-pico/index.html

**Comment by michi-jung** [comment]
Hi @acassis 

Thanks a lot for your review.

I did open https://github.com/apache/nuttx-apps/pull/2361 for the printf-style format string errors in dhcp6c.c

The board documentation is in https://github.com/apache/nuttx/pull/12186/commits/54e61ca08da87ab52597ca2a27a47488786b8d7d

I will handle your other remarks.

File: boards/arm/rp2040/w5500-evb-pico/include/board.h (hunk 1 above)
Comment (michi-jung) [reply boards/arm/rp2040/w5500-evb-pico/include/board.h:151]: Thanks! Done.

File: boards/arm/rp2040/w5500-evb-pico/include/board.h (hunk 2 above)
Comment (michi-jung) [reply boards/arm/rp2040/w5500-evb-pico/include/board.h:162]: Thanks! Done.

File: boards/arm/rp2040/w5500-evb-pico/include/rp2040_i2cdev.h (hunk 3 above)
Comment (michi-jung) [reply boards/arm/rp2040/w5500-evb-pico/include/rp2040_i2cdev.h:59]: Thanks! Done.

**Comment by acassis** [comment]
> Hi @acassis
> 
> Thanks a lot for your review.
> 
> I did open [apache/nuttx-apps#2361](https://github.com/apache/nuttx-apps/pull/2361) for the printf-style format string errors in dhcp6c.c
> 
> The board documentation is in [182c02a](https://github.com/apache/nuttx/commit/182c02a71486350b4d24705891d2e75320c4bfe0)
> 
> I will handle your other remarks.

Thank you @michi-jung! Please convert the README.txt to ReStructuredText to include pictures. Those README.txt were moved to Documentation/ by @raiden00pl to be fixed/converted later, but he didn't finish the convertion yet.

So it is causing confusion, you are the second person to submit an old style README.txt instead of using the new format.

File: boards/arm/rp2040/common/src/rp2040_w5500.c [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+ ****************************************************************************/
+
+static struct rp2040_lower_s g_enclower =
+{
+  .lower =
+  {
+    .frequency = (CONFIG_RP2040_W5500_SPI_FREQ * 1000),
+    .spidevid  = 0,
+    .mode      = SPIDEV_MODE0,
+    .attach    = up_attach,
```
Comment (xiaoxiang781216) [inline boards/arm/rp2040/common/src/rp2040_w5500.c:87]: should we name up_ to rp2040_

Comment (michi-jung) [reply boards/arm/rp2040/common/src/rp2040_w5500.c]: Good idea.  Done.

**Comment by michi-jung** [comment]
> > Hi @acassis
> > Thanks a lot for your review.
> > I did open [apache/nuttx-apps#2361](https://github.com/apache/nuttx-apps/pull/2361) for the printf-style format string errors in dhcp6c.c
> > The board documentation is in [182c02a](https://github.com/apache/nuttx/commit/182c02a71486350b4d24705891d2e75320c4bfe0)
> > I will handle your other remarks.
> 
> Thank you @michi-jung! Please convert the README.txt to ReStructuredText to include pictures. Those README.txt were moved to Documentation/ by @raiden00pl to be fixed/converted later, but he didn't finish the convertion yet.
> 
> So it is causing confusion, you are the second person to submit an old style README.txt this week instead of using the new format.

Ok, I basically integrated the contents of the README.txt file into the ReStructuredText file.  Thanks for your support, @acassis

**Review by acassis — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**